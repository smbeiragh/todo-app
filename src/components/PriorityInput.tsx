import * as React from 'react';
import classnames from 'classnames';

const priorityNames = ['low', 'medium', 'high'];

type Props = {
  onChange: (value: number) => void,
  value: number,
  min: number,
  max: number,
  className: string
}

type DefaultProps = {
  onChange: (value: number) => void,
  value: number,
  min: number,
  max: number,
  className: string
}

export class PriorityInput extends React.Component<Props> {

  static defaultProps: DefaultProps = {
    onChange: ()=>{},
    value: 1,
    min: 0,
    max: 2,
    className: ''
  }

  constructor(props: Props) {
    super(props);
  }

  handleChange(delta: number) {
    const { value, onChange, min, max } = this.props;
    const newValue = value + delta;
    if(newValue <= max && newValue >= min) {
      onChange(newValue)
    }
  }

  render() {
    const { value, className, min, max } = this.props;
    return (
      <div className={classnames(className, 'c-priorityInput')}>
        <button
          className={'c-priorityInput__btn c-priorityInput__btn--up'}
          onClick={(e) => { e.preventDefault(); this.handleChange(1); }}
          disabled={value === max}
        >
          <i className={'c-priorityInput__icon icon-less'}></i>
        </button>
        <div className={'c-priorityInput__display'} >{ priorityNames[value] }</div>
        <button
          className={'c-priorityInput__btn c-priorityInput__btn--down'}
          onClick={(e) => { e.preventDefault(); this.handleChange(-1);}}
          disabled={value === min}
        >
          <i className={'c-priorityInput__icon icon-more'}></i>
        </button>
      </div>
    );
  }
}