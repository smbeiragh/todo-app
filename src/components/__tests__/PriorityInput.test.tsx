import * as React from 'react';
import { PriorityInput } from '../PriorityInput';
import * as renderer from 'react-test-renderer';
import {shallow} from "enzyme";

describe('PriorityInput', () => {
  it('click on buttons should trigger onChange', () => {
    const clickFn = jest.fn();
    const component = shallow(<PriorityInput onChange={clickFn} value={1} />);
    component
      .find('.c-priorityInput__btn--up')
      .simulate('click', {
        preventDefault: () => {
        }
      });
    expect(clickFn).toHaveBeenCalled();
  });

  it('PriorityInput should render', () => {
    const component = renderer.create(
      <PriorityInput value={1} onChange={()=>{}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
