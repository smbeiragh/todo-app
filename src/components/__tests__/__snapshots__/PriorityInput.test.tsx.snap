// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriorityInput PriorityInput should render 1`] = `
<div
  className="c-priorityInput"
>
  <button
    className="c-priorityInput__btn c-priorityInput__btn--up"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="c-priorityInput__icon icon-less"
    />
  </button>
  <div
    className="c-priorityInput__display"
  >
    medium
  </div>
  <button
    className="c-priorityInput__btn c-priorityInput__btn--down"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="c-priorityInput__icon icon-more"
    />
  </button>
</div>
`;
